src/Roslyn.Diagnostics.Analyzers/Core/PartsExportedWithMEFv2MustBeMarkedAsShared.Fixer.cs

```diff
@@ -1,29 +1,66 @@
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
 
 namespace Roslyn.Diagnostics.Analyzers
 {
     /// <summary>
     /// RS0023: Parts exported with MEFv2 must be marked as Shared
     /// </summary>
-    public abstract class PartsExportedWithMEFv2MustBeMarkedAsSharedFixer : CodeFixProvider
+    public abstract class PartsExportedWithMEFv2MustBeMarkedAsSharedFixer<TTypeSyntax> : CodeFixProvider
+        where TTypeSyntax : SyntaxNode
     {
-        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray<string>.Empty;
+        public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(RoslynDiagnosticIds.MissingSharedAttributeRuleId);
 
         public sealed override FixAllProvider GetFixAllProvider()
         {
             // See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
             return WellKnownFixAllProviders.BatchFixer;
         }
 
-        public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
         {
-            // Fixer not yet implemented.
-            return Task.CompletedTask;
+            var document = context.Document;
+            var cancellationToken = context.CancellationToken;
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            foreach (var diagnostic in context.Diagnostics)
+            {
+                // The following doesn't seem to work correctly due to a possible Roslyn bug.
+                //var symbol = semanticModel!.GetEnclosingSymbol(diagnostic.Location.SourceSpan.Start);
+                //if (symbol.DeclaringSyntaxReferences.IsEmpty)
+                //{
+                //    continue;
+                //}
+                //var declaration = await symbol.DeclaringSyntaxReferences[0].GetSyntaxAsync(cancellationToken).ConfigureAwait(false);
+
+                if (TryGetDeclaration(root!.FindNode(diagnostic.Location.SourceSpan), out var declaration))
+                {
+                    context.RegisterCodeFix(
+                        CodeAction.Create(
+                            RoslynDiagnosticsAnalyzersResources.AddSharedAttribute,
+                            _ => AddSharedAttribute(document, root!, declaration),
+                            equivalenceKey: nameof(RoslynDiagnosticsAnalyzersResources.AddSharedAttribute)),
+                        diagnostic);
+                }
+            }
+        }
+
+        private static bool TryGetDeclaration(SyntaxNode node, [NotNullWhen(true)] out SyntaxNode? declaration)
+            => (declaration = node.FirstAncestorOrSelf<TTypeSyntax>()) is not null;
```

Comment:
💡 In might be possible to write this without needing a separate abstract base class/derived classes. If you wanted to try this (which could certainly be a follow-up pull request), you could look at `ExportedPartsShouldHaveImportingConstructorCodeFixProvider.AddImportingConstructorAttributeAsync`.